Docs: API Doc hackathon project (#209)
grafana/writers-toolkit: docs/sources/writing-guide/contribute-documentation/index.md

> **Note:** You can only edit a topic that is part of the latest release of documentation.

### Before you begin

- Create a [GitHub](https://www.github.com) account.

**To edit a topic:**

1. From a topic on the documentation website, click **Edit this page** (pencil icon).
1. Enter your changes.
1. Change the branch name, if required.
   The branch name is auto-populated.

1. Click **Propose changes**.

   A PR is created and then goes through the PR review and approval workflow.

## Develop a new topic

> If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the technical writing team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.

Reviewer: ```suggestion
If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the Docs team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.
```

Author: Fiona preferred the other way.

> **Note:** You can only edit a topic that is part of the latest release of documentation.

### Before you begin

- Create a [GitHub](https://www.github.com) account.

**To edit a topic:**

1. From a topic on the documentation website, click **Edit this page** (pencil icon).
1. Enter your changes.
1. Change the branch name, if required.
   The branch name is auto-populated.

1. Click **Propose changes**.

   A PR is created and then goes through the PR review and approval workflow.

## Develop a new topic

> If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the Docs team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.

Reviewer 2: "create a documentation plan"  I'm worried this will discourage community members and developers from contributing to the documentation.  Even as a technical writer, I don't write a content plan for every topic I write.  Other problems with this introduction:  1) How would they know how to contact the docs team?  2) this content assumes that the docs team has the bandwidth to consult on doc plans.

Author: All good points. Do you have an alternative advice for developers who want to create a new topic? Are you saying we don't want them to consult a writer at the outset of a process but to wait until it's already been drafted before reaching out to us?

Also the contact info for the doc team is given elsewhere in the doc. We could add it here, too.

Reviewer 2: As much as I'd love to be consulted about everything people want to write?  That makes me the bottleneck for documentation.  I'd rather content get written if users need it.  We can always iterate on it later.

Author: It's true that there is an additional time ask of the technical writer: to look at the developer's doc plan and give feedback/advice based on the outline. However, I would make the argument that there is potentially much more time saved later on by by having fewer major corrections to make... and perhaps improvements in document quality. Personally, I would want to be consulted for.a quick sanity check before a big doc gets written, but I see that there is an argument for a different process to be made.

Reviewer: I had a conversation with @krajorama today about where to put stuff for Helm chart documentation, and that is an example of far less work due to an up-front conversation. The earlier the ping the better, imo.

Author: deleted from this PR, will submit new PR in the future

> If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the Docs team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.

Reviewer 3: ```suggestion
If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the technical writing team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.
```

> If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the Doc Squad to see it through. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.

Reviewer: ```suggestion
If you want to develop a new topic from scratch, you can create a documentation plan and collaborate with a member of the Docs team. According to the book [_Docs for Developers_](https://docsfordevelopers.com/), a documentation plan is a _flexible outline_ for anticipating where the writing process will lead you.
```

> Your documentation plan will help you to:

Reviewer: ```suggestion
Your documentation plan helps you to:
```

> - Identify existing information gaps and explain how you will fill them
> - Get feedback from users and stakeholders before the writing process begins
> - Consider different approaches you might take, and decide between them

Reviewer: ```suggestion
- Identify existing information gaps, and explain how you will fill them.
- Get feedback from users and stakeholders before the writing process begins.
- Consider different approaches you might take, and decide on one of them.
```

> > **Note:** Only create new topics that are part of the upcoming release of documentation.

Reviewer 2: This is a weird restriction to place on anyone writing new content.  Once the release is out, the feature can never be documented?   Again, I am worried that this will discourage documentation contributions. There are PLENTY of places where released features and code still needs to be documented and we can use all the help we can get.

Author: Right, it's not the best advice. I've deleted it in the most recent version of this PR.

> > **Note:** You can only create new topics that are part of the latest release of documentation.

Reviewer: What does this mean? I can create new topics for the `next` release of documentation. I rarely touch `latest` documentation.

Reviewer: Do you mean "upcoming"?

Reviewer: Technically, I can create new topics for any release of documentation, but it is not a good idea. :)

Author: I will delete the note.

> > **Note:** You can only create new topics that are part of the latest release of documentation.

Reviewer: ```suggestion
> **Note:** Only create new topics that are part of the upcoming release of documentation.
```

**To develop a new topic:**

> 1. Create a new document in Google Docs, Google Sheets, or a similar tool. For a small project, you may use an email message.

Reviewer: ```suggestion
1. Create a new document in Google Docs, Google Sheets, or a similar tool.
```
Simplified instruction.

1. Copy and paste content from the [Document Planning Template]({{< relref "../../static/templates/document-planning-template.md) into your document. 
1. Fill in the requested information. 
1. Attach an outline of your proposed document. 

> 1. Schedule a review meeting with a member of the Doc Squad. 

Reviewer: ```suggestion
1. Schedule a review meeting with a member of the Docs team.
```

>    If you don't know which writer will be reviewing your documentation plan, reach out to [docs@grafana.com](mailto:docs@grafana.com). 

Reviewer 2: Fiona is putting boundaries around what the team can and cannot do with our current staffing, but this makes it sound like we have the bandwidth to collaborate on any docs requests that come in.

Author: This is definitely a valid concern. One way to resolve it would be to delete the content about "developing a new topic" entirely. Another way would be to add caveats to the effect that all requests will be triaged.

>    If you don't know which writer will be reviewing your documentation plan, reach out to [docs@grafana.com](mailto:docs@grafana.com).  

Reviewer: ```suggestion
   If you don't know which writer will be reviewing your documentation plan, reach out to [docs@grafana.com](mailto:docs@grafana.com). 
```
@GrafanaWriter, should this be `docs-team@grafana.com` instead now?

> Your writer on the Doc Squad will contact you with next steps. Our aim is to help you to be successful in improving the experience of our users and developers.

Reviewer: ```suggestion
Your writer on the Docs team will contact you with next steps. Our aim is to help you to be successful in improving the experience of our users and developers.
```

1. Commit your changes.
1. Push your changes to Github.
1. Create a PR in Github.

> 1. Add the `type/docs` label.

Reviewer: I prefer to keep this rationale in here. Alternatively, and below this line, we could add instead:

"The Docs team uses the `type/docs` label across all Grafana Labs project and products to manage incoming documentation issues."

> 1. Add the `type/docs` label.

Author: Weird. As with the other case, I didn't (at least not intentionally). I have added it back and will push it soon.

When you edit the `main` branch of a project, it affects the content in the `next` directory of the website.
To edit a previous version, or `latest` (the most recent release), you must backport the changes into the long-lived version branches in the project repository.

> To backport a change, use the `backport <vMAJOR.MINOR.x>` labels on the GitHub pull request.

Reviewer 2: Why did you remove the link that points out that we have an entire other topic about backporting?  (I'm seeing you remove a lot of updates that I have recently inserted because I thought they were needed...)

Author: Again, I didn't do so intentionally. Are you sure your additions were merged into the live repo?

Author: I know, it seems strange to even ask since we can see them. But I didn't delete these.
